docarray/array/storage/sqlite/binary.py

```diff
@@ -0,0 +1,134 @@
+from typing import Union, BinaryIO, TYPE_CHECKING, Type, Optional, Generator
+from docarray.array.mixins import BinaryIOMixin
+
+if TYPE_CHECKING:
+    from ....types import T
+    from .... import Document, DocumentArray
+
+
+def _check_protocol(protocol):
+    if protocol == 'pickle-array':
+        raise ValueError(
+            'protocol pickle-array is not supported for DocumentArraySqlite'
+        )
+
+
+class SqliteBinaryIOMixin(BinaryIOMixin):
+    """Save/load an array to a binary file."""
+
+    @classmethod
+    def load_binary(
+        cls: Type['T'],
+        file: Union[str, BinaryIO, bytes],
+        protocol: str = 'protobuf-array',
+        compress: Optional[str] = None,
+        _show_progress: bool = False,
+        streaming: bool = False,
+    ) -> Union['DocumentArray', Generator['Document', None, None]]:
+        """Load array elements from a compressed binary file.
+
+        :param file: File or filename or serialized bytes where the data is stored.
+        :param protocol: protocol to use. 'pickle-array' is not supported for DocumentArraySqlite
+        :param compress: compress algorithm to use
+        :param _show_progress: show progress bar, only works when protocol is `pickle` or `protobuf`
+        :param streaming: if `True` returns a generator over `Document` objects.
+        In case protocol is pickle the `Documents` are streamed from disk to save memory usage
+        :return: a DocumentArray object
+        """
+        _check_protocol(protocol)
```

Comment:
i think a better way to do this is the following:
1. moving `_check_protocol(protocol)` to `DocumentArray` level mixin
2. add class variable `available_protocols `, e.g.
```python
class IOMixin:

   available_protocols = ['a', 'b', 'c', 'd']
   def _check_protocols(self, value: str):
       if value not in self.available_protocols:
          raise ...

   def from_bytes():
      self._check_protocols

```
3. Finally, all DA subclass just need to specify a white list by overriding `available_protocols` class variable

Reply:
@hanxiao I'll still need to subclass the mix and override methods like `load_binary` because I need to change the default protocol. Is it okay to implement your suggestion while still overriding the methods ?

There's another option which is adding attribute `default_protocol` in the same way, and modifying the protocol parameter to be None (will bad for autocompletion).
Your thoughts ?